Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +914,76 @@ export class OtherSettings {
+        }
+        const setting = section.inspect<boolean>(settingName);
+        return setting?.defaultValue ?? defaultValue;
+    }
 
-    public getCustomColorToken(colorTokenName: string): string | undefined { return vscode.workspace.getConfiguration("editor.tokenColorCustomizations").get<string>(colorTokenName); }
-    public getCustomThemeSpecificColorToken(themeName: string, colorTokenName: string): string | undefined { return vscode.workspace.getConfiguration(`editor.tokenColorCustomizations.[${themeName}]`, this.resource).get<string>(colorTokenName); }
+    private getAsNumber(sectionName: string, settingName: string, resource: any, defaultValue: number, minimum?: number, maximum?: number): number {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<number>(settingName);
+        // Validates the value is a number and clamps it to the specified range. Allows for undefined maximum or minimum values.
+        if (isNumber(value)) {
+            if (minimum !== undefined && value < minimum) {
+                return minimum;
+            }
+            if (maximum !== undefined && value > maximum) {
+                return maximum;
+            }
+            return value;
+        }
+        const setting = section.inspect<number>(settingName);
+        return setting?.defaultValue ?? defaultValue;
+    }
 
-    public get customTextMateRules(): TextMateRule[] | undefined { return vscode.workspace.getConfiguration("editor.tokenColorCustomizations").get<TextMateRule[]>("textMateRules"); }
-    public getCustomThemeSpecificTextMateRules(themeName: string): TextMateRule[] | undefined { return vscode.workspace.getConfiguration(`editor.tokenColorCustomizations.[${themeName}]`, this.resource).get<TextMateRule[]>("textMateRules"); }
+    private getAsAssociations(sectionName: string, settingName: string, resource?: any): Associations {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<any>(settingName);
+        if (isValidMapping(value, 'string', 'string')) {
+            return value as Associations;
+        }
+        const setting = section.inspect<any>(settingName);
+        return setting?.defaultValue as Associations;
+    }
+
+    private getAsExcludes(sectionName: string, settingName: string, resource?: any): Excludes {
+        const section = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = section.get<any>(settingName);
+        if (isValidMapping(value, 'string', 'boolean')) {
+            return value as Excludes;
+        }
+        const setting = section.inspect<any>(settingName);
+        return setting?.defaultValue as Excludes;
+    }
+
+    // All default values are obtained from the VS Code settings UI. Please update the default values as needed.
+    public get editorTabSize(): number { return this.getAsNumber("editor", "tabSize", this.resource, 4, 1); }
```

Comment:
I see code checking this for `null`. Same with editorInsertSpaces. Is that checking code valid or what?

Reply:
Can you show me the null checks? I don't see them.

Reply:
Oh, it's gone now. You may have changed that in your last commit?